lib/json/serialization.nit

```diff
@@ -448,6 +448,26 @@ class JsonDeserializer
 	end
 end
 
+redef class Text
+
+	# Deserialize a `nullable Object` from this JSON formatted string
+	#
+	# Warning: Deserialization errors are reported with `print_error` and
+	# may be returned as a partial object or as `null`.
+	#
+	# This method is not appropriate when errors need to be handled programmatically,
+	# manually use a `JsonDeserializer` in such cases.
+	fun from_json_string: nullable Object
+	do
+		var deserializer = new JsonDeserializer(self)
+		var res = deserializer.deserialize
+		if deserializer.errors.not_empty then
+			print_error "Deserialization Errors: {deserializer.errors.join(", ")}"
```

Comment:
Since you just return a nullable Object, then people will have to cast anyway, so, what about returning an Error object, so client that care can test the error, other will fail at the cast.